### PR TITLE
Add @ObservedObject attachments to respond to attachment changes.

### LINKED ISSUE #247: EMS note does not work
When selecting the paper clip in the 'Electronic Monitoring System' section on the 'Vessel' page - if I try to add a note nothing happens. A field should appear where I can write in a note and it should be saved to the appropriate place in Realm.

### PR DESCRIPTION
**Pull request by coltonlemmon**
## Related Issue

Fixes #247 

## Checklist:

- [x] I have read the [contributor's guide](https://wildaid.github.io/contribute/index.html).
- [x] I linked an issue in the previous section
- [x] I have commented on the linked issue
- [x] I was assigned the linked issue (not required)
- [x] I have tested the change to the best of my ability against the [sandbox](https://wildaid.github.io/contribute/sandbox.html) or a [local build](https://wildaid.github.io/build).

Optional items:

- [ ] My change adds new text and requires a change to translations.
- [ ] My change requires a change to the documentation.
- [ ] I have submitted a PR to the [documentation repo](https://github.com/WildAid/wildaid.github.io).
- [ ] I was not able to test... (explain below, e.g. you did not have permissions to test a specific feature)
- [ ] This change depends O-FISH Realm repository changes (explain below)
- [ ] This change depends O-FISH Web repository changes (explain below)

* **Optional: Add any explanations here** 
Modifying attachments wasn't updating the correct view because the EMSInputView was not observing AttachmentsViewModel directly. There may be a better way to do this that I am unaware of.

Also note that there is a separate issue when attempting to delete a note. I'm going to create an issue for this.

* **Optional: Add any relevant screenshots here**

### CODE REVIEW

**Comment by am-MongoDB** [comment]
It looks like the problem is wider than just notes for EMS - impacts other views too

File: o-fish-ios.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2039,7 +2039,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 1;
 				DEVELOPMENT_ASSET_PATHS = "\"o-fish-ios/Preview Content\"";
-				DEVELOPMENT_TEAM = PH747F3726;
+				DEVELOPMENT_TEAM = P3B9MPU7J2;
```
Comment (am-MongoDB) [inline o-fish-ios.xcodeproj/project.pbxproj:2042]: Don't make this change

File: o-fish-ios.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2063,7 +2063,7 @@
...
```
Comment (am-MongoDB) [inline o-fish-ios.xcodeproj/project.pbxproj:2066]: Don't make this change

File: o-fish-ios/Views/ReportFlow/Vessel/EMSInputView.swift [hunk 3]
```diff
@@ -9,6 +9,7 @@ import SwiftUI
 
 struct EMSInputView: View {
     @ObservedObject var ems: EMSViewModel
+    @ObservedObject var attachments: AttachmentsViewModel
```
Comment (am-MongoDB) [inline o-fish-ios/Views/ReportFlow/Vessel/EMSInputView.swift:12]: This problem isn't limited to just this view - it's anywhere that a note is attached. Would be good to find a solution that's in `AddAttachmentsButton` or lower

**Review by am-MongoDB — CHANGES_REQUESTED** [review]

**Comment by am-MongoDB** [comment]
I think I may have a solution to this as part of working on https://github.com/WildAid/o-fish-ios/issues/368

**Comment by am-MongoDB** [comment]
This has been fixed under https://github.com/WildAid/o-fish-ios/pull/375 along with the crash when deleting notes.

The solution I opted for was to remove the "emptiness" tests surrounding the embedding of `AttachmentView`, as this preventing the AttachmentView being created even after a note was added – this shouldn't be needed if everything in SwiftUI was working as advertised.

Thanks @coltonlemmon for your work on this!

**Comment by coltonlemmon** [comment]
@am-MongoDB Thats a good solution. I had a hard time understanding what was wrong because it seemed like it should have been working correctly to me. 

If you found this PR helpful would you mind marking it as hacktoberfest-accepted?

**Pull request closed without merging**